### ISSUE TITLE
[Lua] Type Error: "this.SpawnXY is Not A Function" In A Specific Level

### ISSUE THREAD

**Issue by commandhat** [codecombat/codecombat#4679]
![chrome_2018-05-05_16-23-37](https://user-images.githubusercontent.com/3821018/39667656-c5532714-5080-11e8-8a3a-9d8569327eea.png)

Level in question: [Game Development, "Them Bones"](https://codecombat.com/play/level/them-bones?)

It appears this level was not created with lua in mind. I edited the preset code to be lua compatible, here is how it looks when I press Restart:

>// Generators spawn enemies over time.
>// Skeletons are afraid of lightstones.
>
>var player = game.spawnPlayerXY("champion", 15, 35);
>player.attackDamage = 60;
>player.maxSpeed = 8;
>
>game.addSurviveGoal();
>game.addDefeatGoal();
>game.spawnXY("x-mark-stone", 60, 35);
>// Spawn a "generator"
>
>// Spawn a "lightstone"
>
>// Now beat your game!

Here is how the code _should_ look, using the Lua language:

>-- Generators spawn enemies over time.
>-- Skeletons are afraid of lightstones.
>
>local player = game.spawnPlayerXY("champion", 15, 35)
>player.attackDamage = 60
>player.maxSpeed = 8
>
>game.addSurviveGoal()
>game.addDefeatGoal()
>game.spawnXY("x-mark-stone", 60, 35)
>-- Spawn a "generator"
>
>-- Spawn a "lightstone"
>
>-- Now beat your game! 

Information from FAQ:
OS: Windows 10
Browser: Google Chrome
Has It Worked Before: Unsure, as this is the first time I reached this level.

**Comment by nwinter** [comment]
@commandhat As Lua and CoffeeScript are somewhat experimental languages for us, we don't have sample code for all levels in those languages. I'd love your help in adding sample code to any levels where you don't see Lua show up if you're willing to help? Here's a quick video I made showing how to do it:

https://www.dropbox.com/s/arq5e6ng5az2top/Adding%20Lua%20sample%20code%20in%20the%20level%20editor.mp4?dl=0

Basically, you open https://codecombat.com/editor/level/them-bones, double-click on the hero placeholder, find the Programmable Component, find the plans entry, open languages, and add/edit the sample code for the target language (here Lua) by copying from the Python and making the appropriate changes. When done, you can submit a patch by clicking the save icon in the upper right of the level editor.

**Comment by commandhat** [comment]
Gotcha. Whenever I have spare time, I'll try to improve CodeCombat in this way.
